test/test_sparse_ops.py

```diff
@@ -0,0 +1,47 @@
+import math
```

Comment:
This can be put into test_sparse.py, since each test file can have multiple test classes.

Reply:
This is done, please check the new implementation in test_sparse.py